### PR TITLE
Show close button in tipping banner

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/26744
https://github.com/brave/brave-browser/issues/26739
https://github.com/brave/brave-browser/issues/26740

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Lots space resolved in this, check in tablet
2. Close button will be top left 
3. "Support this creator" text 

image

image

### CODE REVIEW

File: android/java/res/layout/brave_rewards_site_banner.xml [hunk 1]
```diff
@@ -36,4 +36,15 @@
...
\ No newline at end of file
+    <ImageView
+        android:id="@+id/button_close"
+        android:layout_width="30dp"
+        android:layout_height="30dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginTop="20dp"
+        android:layout_marginEnd="20dp"
+        android:src="@drawable/ic_baseline_close_24"
```
Comment (deeppandya) [inline android/java/res/layout/brave_rewards_site_banner.xml:47]: We should use app:srcCompat else it would crash on Android 6

Comment (sujitacharya2005) [reply android/java/res/layout/brave_rewards_site_banner.xml:48]: used both android:src=  and app:srcCompat

Comment (deeppandya) [reply android/java/res/layout/brave_rewards_site_banner.xml:48]: > used both android:src=  and app:srcCompat
> 

Actually we don't need to use android:sec at all as app:srcCompat also supports  backward compatible.

Comment (sujitacharya2005) [reply android/java/res/layout/brave_rewards_site_banner.xml:48]: https://github.com/chromium/chromium/blob/main/chrome/browser/notifications/android/java/res/layout/notification_permission_rationale_dialog.xml

**Review by deeppandya — APPROVED** [review]
LGTM

**Comment by kjozwiak** [comment]
## `Test Case #1` - https://github.com/brave/brave-core/pull/16081

Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.48.24 Chromium: 108.0.5359.71 (Official Build) canary (32-bit)
--- | ---
Revision | 1e0e3868ee06e91ad636a874420e3ca3ae3756ac-refs/branch-heads/5359@{#1016}
OS | Android 13; Build/TQ1A.221205.011
```

* ensured that pressing the `X` button closes the tipping banner without any issues on both one-time & monthly tipping
* ensured that you can still close/dismiss the tipping banner by swiping to the left and returning to the rewards panel

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221208-144337 | Screenshot_20221208-144342

--------------------------

Verification **PASSED** on `Samsung Galaxy Tab S8 Ultra` running `Android 13` using the following build(s):

```
Brave | 1.48.24 Chromium: 108.0.5359.71 (Official Build) canary (32-bit)
--- | ---
Revision | 1e0e3868ee06e91ad636a874420e3ca3ae3756ac-refs/branch-heads/5359@{#1016}
OS | Android 13; Build/TP1A.220624.014
```

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221208_145312_Brave - Nightly | Screenshot_20221208_145320_Brave - Nightly

* ensured that pressing the `X` button closes the tipping banner without any issues on both one-time & monthly tipping
* ensured that you can still close/dismiss the tipping banner by tapping on the `<` (Back) button at the bottom of the device

## `Test Case #2` - https://github.com/brave/brave-browser/issues/26739

Verification **PASSED** on `Samsung Galaxy Tab S8 Ultra` running `Android 13` using the following build(s):

```
Brave | 1.48.24 Chromium: 108.0.5359.71 (Official Build) canary (32-bit)
--- | ---
Revision | 1e0e3868ee06e91ad636a874420e3ca3ae3756ac-refs/branch-heads/5359@{#1016}
OS | Android 13; Build/TP1A.220624.014
```

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221208_145312_Brave - Nightly | Screenshot_20221208_145320_Brave - Nightly

## `Test Case #3` - https://github.com/brave/brave-browser/issues/26739

Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.48.24 Chromium: 108.0.5359.71 (Official Build) canary (32-bit)
--- | ---
Revision | 1e0e3868ee06e91ad636a874420e3ca3ae3756ac-refs/branch-heads/5359@{#1016}
OS | Android 13; Build/TQ1A.221205.011
```

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221208-144337 | Screenshot_20221208_145312_Brave - Nightly

* esnsured that `Support this Creator` is being displayed via the tipping panel

**Pull request merged**